plugins/modules/setup.ps1

```diff
@@ -559,17 +559,29 @@ $factMeta = @(
     @{
         Subsets = 'all_ipv4_addresses', 'all_ipv6_addresses'
         Code = {
-            $interfaces = [System.Net.NetworkInformation.NetworkInterface]::GetAllNetworkInterfaces()
-            $ips = @(foreach ($interface in $interfaces) {
-                    # Win32_NetworkAdapterConfiguration did not return the local IPs so we replicate that here.
-                    $interface.GetIPProperties().UnicastAddresses | Where-Object {
-                        $_.Address.ToString() -notin @('::1', '127.0.0.1')
-                    } | ForEach-Object -Process {
-                        $_.Address.ToString()
+            $NetIPAddress = Get-CimInstance -Namespace ROOT/StandardCimv2 -ClassName MSFT_NetIPAddress
```

Comment:
We are purposefully avoiding WMI/CIM in this module where we can as it won't work when a user is running as a non-admin. The data returned by the dotnet methods should contain enough information in the `$_.Address` object that tells you whether it's an IPv4 vs IPv6 address. As for why it's returning extra entries compared to the CIM method that might be something you need to investigate on your host to figure out what those extra entries are from and whether we should filter them out or not.